### PR TITLE
Add support for multiple (un)deploy sounds on GrantConditionOnDeploy

### PR DESCRIPTION
**Pull request by abcdefg30**

### CODE REVIEW

**Review by reaperrr — DISMISSED** [review]

**Comment by abcdefg30** [comment]
Rebased.

File: OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs [hunk 1]
```diff
@@ -221,8 +221,8 @@ void Deploy(bool init)
 			if (!IsValidTerrain(self.Location))
 				return;
 
-			if (!string.IsNullOrEmpty(Info.DeploySound))
-				Game.Sound.Play(SoundType.World, Info.DeploySound, self.CenterPosition);
+			if (Info.DeploySounds != null && Info.DeploySounds.Any())
+				Game.Sound.Play(SoundType.World, Info.DeploySounds.Random(self.World.SharedRandom), self.CenterPosition);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs]: Sounds aren't synced, so please use `World.LocalRandom`.

**Review by pchote — COMMENTED** [review]
Just one nit, otherwise LGTM.

**Comment by abcdefg30** [comment]
Updated.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/c7e0e1d8c8fcb97496623d68e9db2aba14ea58c5

**Comment by pchote** [comment]
I just noticed that this put the update rule in the wrong directory. Will need a followup to fix this.

**Comment by abcdefg30** [comment]
#16301

**Pull request merged**